packages/flutter_tools/lib/src/base/logger.dart

```diff
@@ -221,7 +221,11 @@ class AnsiTerminal {
   String clearScreen() => supportsColor ? _clear : '\n\n';
 
   set singleCharMode(bool value) {
-    stdin.lineMode = !value;
+    try {
+      stdin.lineMode = !value;
+    } on IOException {
+      // This can throw for some terminals; we ignore the error.
```

Comment:
Yeah, no, thinking about this, this is worrying, and possibly related to the issues @pq is seeing with the `1.22.0-dev.5.0` analysis server locally.